src/backends/backend/python/python_interpreter.cc

```diff
@@ -0,0 +1,902 @@
+      TRITONSERVER_ERROR_INVALID_ARG,
+      std::string("'python_interpreter' backend only supports CPU instances"));
+
+  RETURN_IF_ERROR(instance_state->CreatePythonInterpreter());
+
+  return nullptr;  // success
+}
+
+TRITONSERVER_Error*
+TRITONBACKEND_ModelInstanceExecute(
+    TRITONBACKEND_ModelInstance* instance, TRITONBACKEND_Request** requests,
+    const uint32_t request_count)
+{
+  ModelInstanceState* instance_state;
+  RETURN_IF_ERROR(TRITONBACKEND_ModelInstanceState(
+      instance, reinterpret_cast<void**>(&instance_state)));
+
+  std::vector<TRITONBACKEND_Response*> responses;
+  responses.reserve(request_count);
+
+  for (uint32_t r = 0; r < request_count; ++r) {
+    TRITONBACKEND_Request* request = requests[r];
+
+    TRITONBACKEND_Response* response;
+    RETURN_IF_ERROR(TRITONBACKEND_ResponseNew(&response, request));
+    responses.push_back(response);
+  }
+
+  ModelState* model_state = instance_state->StateForModel();
+
+  bool supports_batching_initialized = false;
+  bool supports_batching = false;
+
+  if (!supports_batching_initialized) {
+    LOG_IF_ERROR(
+        model_state->SupportsFirstDimBatching(&supports_batching),
+        "failed to determine batching support");
+    supports_batching_initialized = true;
+  }
+
+  for (uint32_t r = 0; r < request_count; ++r) {
+    TRITONBACKEND_Request* request = requests[r];
+    TRITONBACKEND_Response* response = responses[r];
+
+    ni::InferenceBatch input_batch;
+
+    uint32_t requested_input_count = 0;
+    uint32_t requested_output_count = 0;
+    GUARDED_RESPOND_IF_ERROR(
+        responses, r,
+        TRITONBACKEND_RequestInputCount(request, &requested_input_count));
+    GUARDED_RESPOND_IF_ERROR(
+        responses, r,
+        TRITONBACKEND_RequestOutputCount(request, &requested_output_count));
+
+    // Batching all the inputs in a single request
+    uint32_t batch_size = 0;
+    for (size_t iidx = 0; iidx < requested_input_count; ++iidx) {
+      ni::InferenceData* input_tensor = input_batch.add_tensors();
+      int err = instance_state->GetInputTensor(
+          iidx, request, input_tensor, responses, r, batch_size);
+      if (err != nic::ErrorCodes::Success) {
+        LOG_ERROR << "Can't get input tensor\n";
+      }
+    }
+
+    input_batch.set_batch_size(batch_size);
+
+    // Inference request
+    grpc::ClientContext context;
+    ni::InferenceBatch result_batch;
+
+    const auto err = instance_state->stub->InferenceRequest(
+        &context, input_batch, &result_batch);
+
+    if (!err.ok()) {
+      LOG_ERROR << "something went wrong with interfere request\n"
+                << err.error_message();
+    }
+
+    // Time to output.
+    ni::TritonJson::Value outputs;
+    LOG_IF_ERROR(
+        instance_state->model_config.MemberAsArray("output", &outputs),
+        "can't get model outputs");
+
+    for (size_t j = 0; j < requested_output_count; ++j) {
+      // Prepare output buffers.
+      TRITONBACKEND_Output* output;
```

Comment:
The request itself will indicate the name of the outputs it expects, you don't need to look at the model configuration. Look carefully at identity.cc to see how outputs are handled. We are going to have to send over to the python which outputs we want produced so it knows what to return. But whatever it returns we should just return those outputs are they are given to from the python.